apps/sms/services/js/drafts.js

```diff
@@ -58,6 +58,7 @@
       draftIndex.set(draft.threadId, drafts);
 
       this.emit('saved', draft);
+      InterInstanceEventDispatcher.emit('drafts-saved', draft);
```

Comment:
I agree with you about the draft update mechanism and we should apply you idea eventually, but maybe not a safe option since it's only 5 day left before RA. And just like you said we will rewrite this part in draft service in the future.

About the memory usage, I was not aware of that because user seldom perform batch action under more than 1 message instance case, which means they need to pause their new/share activity and switch back to original message app for further actions. Anyway I just profiled with b2g-info but found that the memory didn't increase even batch removing the threads with drafts. I thought the memory should somehow increased and dropped later since it will force triggering the drafts request for new drafts map every time. But the fact is the number of memory PSS/USS didn't change while deleting.